docker/generate_makefile.py

```diff
@@ -4,200 +4,140 @@
-            FUZZER_BENCHMARK_TEMPLATE.format(fuzzer=fuzzer,
-                                             benchmark=benchmark,
-                                             base_tag=BASE_TAG))
-
-    # Generate rules for building/pulling all target/benchmark pairs.
-    all_build_targets = ' '.join(
-        ['build-{0}-{1}'.format(fuzzer, benchmark) for benchmark in benchmarks])
-    all_pull_targets = ' '.join(
-        ['pull-{0}-{1}'.format(fuzzer, benchmark) for benchmark in benchmarks])
-    print('build-{fuzzer}-all: {all_targets}'.format(
-        fuzzer=fuzzer, all_targets=all_build_targets))
-    print('pull-{fuzzer}-all: {all_targets}'.format(
-        fuzzer=fuzzer, all_targets=all_pull_targets))
+            RUN_TEMPLATE.format(run_type=run_type,
+                                benchmark=benchmark,
+                                fuzzer=fuzzer))
+        print('\t-e FUZZ_TARGET=$({benchmark}-fuzz-target) \\'.format(
+            benchmark=benchmark))
+        if run_type == 'test-run':
+            print('\t-e MAX_TOTAL_TIME=20 \\\n\t-e SNAPSHOT_PERIOD=10 \\')
+        if run_type == 'debug':
+            print('\t-entrypoint "/bin/bash" \\\n\t-it ', end='')
+        else:
+            print('\t', end='')
+        print(BASE_TAG + '/' + image['tag'])
+        print()
+
+
+def print_makefile(name, image):
+    """Print makefile section for given image to stdout."""
+    _print_makefile_build_template(name, image)
+    if 'runner' in name and not ('intermediate' in name or 'base' in name):
+        _print_makefile_run_template(image)
 
 
 def main():
-    """Main entry point."""
-    # Output boilerplate used by other templates and generated rules.
-    print(BOILERPLATE)
-
-    # Compute the list of benchmarks.
-    benchmarks = []
-    for benchmark in os.listdir(BENCHMARKS_DIR):
-        benchmark_path = os.path.join(BENCHMARKS_DIR, benchmark)
-        if not os.path.isdir(benchmark_path):
-            continue
-        if os.path.exists(os.path.join(benchmark_path, 'benchmark.yaml')):
-            benchmarks.append(benchmark)
-
-    # Generate the build rules for fuzzer/benchmark pairs.
-    fuzzers = []
-    for fuzzer in os.listdir(FUZZERS_DIR):
-        # Skip non-directory files. These do not represent fuzzers.
-        fuzzer_dir = os.path.join(FUZZERS_DIR, fuzzer)
-        if not os.path.isdir(fuzzer_dir):
-            continue
-
-        generate_fuzzer(fuzzer, benchmarks)
-        fuzzers.append(fuzzer)
-
-    # Generate rules to build all known targets.
+    """Generates Makefile with docker image build rules."""
+    fuzzers = fuzzer_utils.get_fuzzer_names()
+    benchmarks = benchmark_utils.get_all_benchmarks()
+    buildable_images = docker_images.get_images_to_build(fuzzers, benchmarks)
+
+    print('export DOCKER_BUILDKIT := 1')
+
+    # Print oss-fuzz benchmarks property variables.
+    print_benchmark_definition(benchmarks)
+
+    for name, image in buildable_images.items():
+        print_makefile(name, image)
```

Comment:
consider calling this something like `print_rules_for_image`?

Reply:
Done